src/utils/regex.cc

```diff
@@ -25,12 +25,37 @@
 
 #ifndef WITH_PCRE2
 #if PCRE_HAVE_JIT
-#define pcre_study_opt PCRE_STUDY_JIT_COMPILE
+// NOTE: Add PCRE_STUDY_EXTRA_NEEDED so studying always yields a pcre_extra strucure
+// and we can selectively override match limits using a copy of that structure at runtime.
+#define pcre_study_opt PCRE_STUDY_JIT_COMPILE | PCRE_STUDY_EXTRA_NEEDED
 #else
-#define pcre_study_opt 0
+// NOTE: Add PCRE_STUDY_EXTRA_NEEDED so studying always yields a pcre_extra strucure
+// and we can selectively override match limits using a copy of that structure at runtime.
+#define pcre_study_opt PCRE_STUDY_EXTRA_NEEDED
 #endif
 #endif
 
+#ifdef WITH_PCRE2
+class Pcre2MatchContextPtr {
+ public:
+    Pcre2MatchContextPtr()
+        : m_match_context(pcre2_match_context_create(NULL)) {}
+
+		Pcre2MatchContextPtr(const Pcre2MatchContextPtr&) = delete;
```

Comment:
If disallowing copy constructor, disallow copy-assignment operator too?

Reply:
Thanks for pointing this out. Fixed.